### ISSUE TITLE
pytorch-converted/test_softmax_lastdim has test with negative axis

### ISSUE THREAD

**Issue by skottmckay** [onnx/onnx#814]
https://github.com/onnx/onnx/tree/master/onnx/backend/test/data/pytorch-converted/test_softmax_lastdim has a test with a negative axis as input. 

The ONNX spec does not say whether a negative axis is allowed or not. https://github.com/onnx/onnx/blob/master/docs/Operators.md#Softmax

Should the spec be clarified or the test changed?

**Comment by houseroad** [comment]
Actually, this is a good topic for discussion. If we don't support negative axis, it's probably hard for us to specify the last dimension conveniently.

**Comment by linkerzhang** [comment]
@houseroad , we may use symbolic dimension in this case, right? using negative value is confusing somehow given we have symbolic support in onnx.

**Comment by houseroad** [comment]
@linkerzhang I didn't follow. Could you explain how to use symbolic to solve this case (expressing the last dimension in axis) ?

**Comment by gramalingam** [comment]
@houseroad , so a negative axis k corresponds to an axis of "rank+k", is that right? So, "-1" corresponds to the last dimension?

**Comment by houseroad** [comment]
@gramalingam Yes, it works like python negative index.

**Comment by houseroad** [comment]
Let's move the discussion to https://github.com/onnx/onnx/issues/842
